### DISCUSSION TITLE
Too many "logs" in the registry

### DISCUSSION THREAD

**Discussion by lann** [bytecodealliance/registry#177] [General]
Currently we have 3 things named "log": the "registry log", the "operator log", and multiple "package logs", but one of these things is not like the others:

- The "package" and "operator" logs are both hash-linked lists of records with similar overall validation rules, differing primarily in the set of allowed record entry types.
- The "registry" log is a historical account of (the hashes of) every record added to every package and operator log since the registry's inception. The registry commits to the state of this log with "checkpoints" which include the log's length and a hash covering its contents.

This causes (me) confusion when thinking about and discussing the registry's protocols and implementation, and I'd like to rename one of these two different kinds of "log".

The two synonyms for "log" that are most appealing to me are "ledger" and "journal", with "history" as a distant 3rd. We could rename either of the two kinds of logs, e.g. "Registry Journal" or "Package/Operator Ledger".

**Comment by esoterra** [comment]
My vote is  that package and operator logs remain called logs and that other future logs of the same kind (i.e. that are included in the registry map) are also called logs, and that the registry-level log is called a ledger or similar instead to distinguish it as a different kind of thing.

**Reply by lann** [reply]
Yeah I think I would lean that direction too.

**Reply by peterhuene** [reply]
I also lean in this direction, with either "ledger" or "journal" being adequate to describe the overall registry log.

Would "register" be a little too redundant?

**Reply by lann** [reply]
We can register the Registry Register with the Registry Register Registrar.

**Comment by lann** [comment]
I'll propose "Registry Ledger" to describe the top-level transparency log. There may be some unfortunate conflation with cryptocurrency "distributed ledgers", but hopefully the only people who get confused by this are already confused.

**Reply by lann** [reply]
I attempted to implement this rename in code which quickly drove this bee out of my bonnet; "log" is pretty well embedded in database schema and protocol fields, which makes this a pretty big lift just to change terminology that most consumers will never care about...
